Document/0x05b-Basic-Security_Testing.md

```diff
@@ -302,15 +337,112 @@ uid=0(root) gid=0(root) groups=0(root),1004(input),1007(log),1011(adb),1015(sdca
 
 Rooting an emulator is therefore unnecessary; root access can be established with `adb`.
 
-##### Restrictions When Testing on an Emulator
+#### Host-Device Data Transfer
+TODO `adb push pull`, scp. https://developer.android.com/studio/command-line/adb#copyfiles
 
-There are several downsides to using an emulator. You may not be able to test an app properly in an emulator if the app relies on a specific mobile network or uses NFC or Bluetooth. Testing within an emulator is also usually slower, and the testing itself may cause issues.
 
-Nevertheless, you can emulate many hardware characteristics, such as [GPS](https://developer.android.com/studio/run/emulator-commandline.html#geo "GPS Emulation") and [SMS](https://developer.android.com/studio/run/emulator-commandline.html#sms "SMS").
 
-### Testing Methods
 
-#### Manual Static Analysis
+#### Obtaining and Extracting Apps
+##### App Store
+TODO mirrorapk or gplaycli
+
+##### Recovering the App Package from the Device
+###### From Rooted Devices
+TODO via adb pull
+###### From Non-Rooted Devices
+
+This works on both rooted and non-rooted devices:
+
+```
+$ adb shell pm path sg.vp.owasp_mobile.omtg_android
+package:/data/app/sg.vp.owasp_mobile.omtg_android-Q0Y29-poHT2w8912zTxXNw==/base.apk
+
+$ adb pull /data/app/sg.vp.owasp_mobile.omtg_android-Q0Y29-poHT2w8912zTxXNw==/base.apk
+/data/app/sg.vp.owasp_mobile.omtg_android-Q0Y29-poHT2w8912zTxXNw==/base.apk: 1 file pulled. 17.9 MB/s (5801767 bytes in 0.309s)
+```
+
+#### Installing Apps
+
+Use `adb install` to install an APK on an emulator or connected device.
+
+```bash
+adb install path_to_apk
+```
+
+Note that if you have the original source code and use Android Studio, you do not need to do this because Android Studio handles the packaging and installation of the app for you.
+
+#### Information Gathering
+##### Installed Apps
+TODO
+```shell
+$ adb shell pm list packages
+package:com.android.sdm.plugins.connmo
+package:com.google.android.googlequicksearchbox
+package:com.android.providers.calendar
+package:com.android.providers.media
+package:com.google.android.onetimeinitializer
+package:com.google.android.ext.shared
+package:com.android.wallpapercropper
+package:sg.vp.owasp_mobile.omtg_android
+```
+
+
+
+##### App Basic Information
+TODO aapt, dumpsys
+###### Sandbox
+
+```shell
+$ adb shell pm path sg.vp.owasp_mobile.omtg_android
+package:/data/app/sg.vp.owasp_mobile.omtg_android-Q0Y29-poHT2w8912zTxXNw==/base.apk
+```
+
+```shell
+$ adb shell ls /data/data/sg.vp.owasp_mobile.omtg_android
+```
+###### Permissions
+TODO from androidmanifest or aapt dump
+
+```
+aapt dump permissions omtg.apk
+```
+###### Native Libs
```

Comment:
I think this was also described in the Android RE chapter. Maybe we can describe the tools and installation here and in the RE chapter the actual use cases?

Reply:
I agree, see [comment](https://github.com/OWASP/owasp-mstg/pull/1222#discussion-diff-287551713)